python/lsst/ts/observatory/control/mock/base_group_mock.py

```diff
@@ -0,0 +1,254 @@
+# This file is part of ts_observatory_control.
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+
+__all__ = ["BaseGroupMock"]
+
+import types
+import asyncio
+
+from lsst.ts import salobj
+
+LONG_TIMEOUT = 30  # seconds
+HEARTBEAT_INTERVAL = 1  # seconds
+CLOSE_SLEEP = 5  # seconds
+
+
+class BaseGroupMock:
+    """ Base Mock for groups of CSCs.
+
+    This is useful for unit testing.
+
+    Parameters
+    ----------
+    components : `list` [`str`]
+        A list of strings with the names of the SAL components that are part
+        of the group.
+    output_only : `list` [`str`]
+        A list of strings with the names of the SAL components that only send
+        telemetry and events, but do not reply to commands.
+
+    """
+
+    def __init__(self, components, output_only=()):
+
+        self._components = components
+
+        self.components = [comp.lower() for comp in self._components]
+
+        self._component_names = []
+
+        self._output_only = output_only if output_only is not None else []
```

Comment:
Please check that the entries in `output_only` exist in `components` in order to catch typos. (I'd probably use the `set` class for that).